apps/rush-lib/src/cli/actions/CheckAction.ts

```diff
@@ -33,6 +34,12 @@ export class CheckAction extends BaseRushAction {
       parameterLongName: '--json',
       description: 'If this flag is specified, output will be in JSON format.'
     });
+    this._verboseFlag = this.defineFlagParameter({
+      parameterLongName: '--verbose',
+      description:
+        'If this flag is specified, long lists of package names will not be truncated. ' +
+        `This has no effect if the ${this._jsonFlag.longName} flag is also specified.`
```

Comment:
The `rush check` report is currently organized **by dependency**, which is geared towards deciding the right version:

*"For each dependency, what versions are we using and where?"*
```
dependency-x
  1.2.3
    - project1
    - project2
    - project3
  ^3.2.1
    - project3
    - project4
```

I have always wanted to introduce a second reporting mode that is **by project**, geared towards manually applying fixes:

*"For each project, what are the problems that I need to fix?"*
```
project1
  dependency-x
    * 43  projects using   ">=1.0.0 <2.0.0 || >=3.0.0 <4.0.0"
    - 5   projects using   "1.2.3"
    - 5   projects using   "^1.0.0"
  dependency-y
    - 17  projects using   "^3.2.1"
    * 1   project using    "1.2.3"

project2
  dependency-y
    * 17  projects using   "^3.2.1"
    - 1   project using    "1.2.3"
```

The choices would be sorted in decreasing order of popularity, with a `*` indicating the version that is currently chosen for that project.


To support different report styles, I think we'd want a parameter like:

```
rush check --report=by-dependency
rush check --report=by-project
```

(Technically the JSON output could become `--report=json`, however @elliot-nelson had a whole thread last fall (was it https://github.com/microsoft/rushstack/pull/2777#issuecomment-939372101 ?) that if I remember right arrived at a consensus that `--json` should be a special standard parameter because it imposes very strict requirements on STDOUT.)

If we added this feature, would `--verbose` still make sense? By which I mean, would the docs still be able to clearly explain how `--verbose` affects different report styles?

Reply:
My recommendation would be to use the git history and the current committed pnpm-lock.yaml to deduce that, e.g. a newly added projet is out of sync and is what should be updated, rather than the rest of the repository.

Reply:
@octogonz - that seems useful. I'll play around with putting that together in a separate PR.

@dmichon-msft - Agreed, that would be a great alternative mode. Maybe a `rush update --fix-dependency-mismatches` flag.